src/Sylius/Bundle/CoreBundle/Provider/CatalogPromotionVariantsProvider.php

```diff
@@ -46,8 +46,12 @@ public function provideEligibleVariants(CatalogPromotionInterface $catalogPromot
 
     private function getVariantsProducts(array $configuration, array $variants): array
     {
+        if (!array_key_exists('variants', $configuration)) {
+            return [];
+        }
```

Comment:
Returning here an empty array can lead to a wrong behaviour because the `$variants` array is ignored. I would be for moving this check to the `foreach` in `provideEligibleVariants`